### PR TITLE
Libstore improvements

### PR DESCRIPTION
**Pull request by tfc**
# Motivation

Random improvements while looking over the code:

- some extra `const`s in loops
- fixing some NarMember constructions (wrong parameter order before)
- fixed some casts

# Context

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

**Comment by edolstra** [comment]
What's the advantage of adding `const` to loops like:
```
for (auto & [name, type] : accessor->readDirectory(path)) {
```

**Comment by tfc** [comment]
> What's the advantage of adding `const` to loops like:

Knowing how `readDirectory` is implemented, it's obvious that changing the items during iteration makes no sense. But that's again more domain knowledge needed. The whole codebase is difficult to get into because one needs so much domain knowledge for everything everywhere. It's good reference hygiene to use as much const as possible, and it helps when reading the code. With every additional `const` somewhere people can relieve their brain from having to track if something's changed.

**Review by tomberek — APPROVED** [review]

File: src/libstore/worker-protocol.cc [hunk 1]
```diff
@@ -31,14 +31,14 @@ std::optional<TrustedFlag> WorkerProto::Serialise<std::optional<TrustedFlag>>::r
 void WorkerProto::Serialise<std::optional<TrustedFlag>>::write(const Store & store, WorkerProto::WriteConn conn, const std::optional<TrustedFlag> & optTrusted)
 {
     if (!optTrusted)
-        conn.to << (uint8_t)0;
+        conn.to << uint8_t(0);
```
Comment (Ericson2314) [inline src/libstore/worker-protocol.cc:34]: How do you feel about parentheses vs curly braces for this?

Comment (tfc) [reply src/libstore/worker-protocol.cc:34]: Use parentheses if you accept implicit conversion, and use curly braces if you want strict type matching. Generally prefer strict and defensive handling of types.

in this case, i want to convert "whatever integer to uint8_t", so it's parentheses.

Comment (Ericson2314) [reply src/libstore/worker-protocol.cc:34]: I would think 

- curly braces work in this case
- curly braces should be preferred basically any time they can be used

If it's a literal it doesn't feel like *convert* whatever integer, it should be "this integer is the right type from the get-go" right?

**Review by Ericson2314 — APPROVED** [review]

**Pull request merged**